src/core/lib/event_engine/posix_engine/posix_engine_listener.h

```diff
@@ -119,9 +126,18 @@ class PosixEngineListenerImpl
    public:
     explicit ListenerAsyncAcceptors(PosixEngineListenerImpl* listener)
         : listener_(listener){};
+
+    void UpdateOnAppendCallback(
+        PosixListenerWithFdSupport::OnPosixBindNewFdCallback on_append) {
+      on_append_ = std::move(on_append);
+    }
+
     void Append(ListenerSocket socket) override {
       acceptors_.push_back(new AsyncConnectionAcceptor(
           listener_->engine_, listener_->shared_from_this(), socket));
+      if (on_append_ != nullptr) {
```

Comment:
nit, optional: `if(on_append_)` also works, since AnyInvocable overrides `operator bool`, and the nullptr comparison is equally synthetic since `on_append_` is not a pointer. I'm not sure if we want to standardize on either syntax.

Reply:
Fixed.